### PR TITLE
Shutdown client resources automatically

### PR DESCRIPTION
**Pull request by vemv**
This is more abstract/handy, as otherwise one has to keep track of more state in one's application.

I verified that these changes work in a real application; starting and stopping custom client resources works as intended (notably, by using `{:nb-io-threads 1 :nb-worker-threads 1}` which makes it easy to track if these are being effectively cycled).

Cheers - V

### CODE REVIEW

**Comment by vemv** [comment]
Uhm, after further testing my solution doesn't work properly - threads leak instaed of being shut down. Even when `destroy-client-resource` is being properly called.

One thing I noticed is that `destroy-client-resource` returns a Future, Even when `deref`ing it, the problem persists.

Any ideas?

**Comment by lerouxrgd** [comment]
>This is more abstract/handy, as otherwise one has to keep track of more state in one's application.

As you have to keep track of some state like the connector and commands anyway, I think it's not really necessary to try to hide the optional client-resources.

Moreover, client-resources can be shared with some other classes and should not be shutdown automatically by celtuce.

**Comment by vemv** [comment]
In the application I was working on I found it really awkward though to pass around a reference to the client-resources from defn to defn. A bit hard to explain without showing the code.

In any case, I was trying to accomplish something fairly simple: have Lettuce create fewer threads (currently it creates quite a lot - not ideal for a laptop). In most libraries one just specifies `{:threads 4}` or such.

Here I have to create an object, keep track of it separately from Celtuce, and then shut it down. Even when I tried to shut it down I don't think I really accompished it (as I observed threads being leaked).

tldr this seems a bit harder / more error-prone than it should be, and therefore might be a good candidate to be abstracted away.

...Maybe optionally, for keeping the Celtuce lib unopinionated.
